### ISSUE TITLE
Formatting of hover tooltips is off in VSCode

### ISSUE THREAD

**Issue by tristan957** [JCWasmx86/mesonlsp#13]
**Describe the bug**
Poor formatting of hover tooltips.

![image](https://github.com/JCWasmx86/Swift-MesonLSP/assets/11936772/68c6a033-3e0c-4f60-8332-884ccb6c588d)

**To Reproduce**
1. Hover a function

**Expected behavior**
Better formatting

**Comment by JCWasmx86** [comment]
How would you expect it to get formatted?

The format is currently:

```
<function_name>

<function_docs>

required positional arguments
optional positional arguments
required kwargs
optional kwargs
```
The optional args are surrounded with `[]`

**Comment by tristan957** [comment]
I think that the way clangd formats things is more readable:

![image](https://github.com/JCWasmx86/Swift-MesonLSP/assets/11936772/6cdab438-dfbf-4d85-8f51-8bd4568de872)

I think the way that Pylance shows optional parameters is pretty good.

![image](https://github.com/JCWasmx86/Swift-MesonLSP/assets/11936772/9e6f9558-b093-4127-a0da-eede6c1fcaa1)

**Comment by tristan957** [comment]
My opinions are subjective though.

**Comment by JCWasmx86** [comment]
What do you think of this?

![image](https://github.com/JCWasmx86/Swift-MesonLSP/assets/63594136/1f26e744-8f1e-4714-a472-cca6f4db26d1)

![image](https://github.com/JCWasmx86/Swift-MesonLSP/assets/63594136/9ded7ddb-45f3-4279-aa60-1d7205c1f159)

I've reordered the fields and improved the formatting. But I really want the documentation to be the first line after the identifier

**Comment by tristan957** [comment]
Does Meson not document what the defaults are for optional kwargs? I still think that the bracketed optionals aren't very intuitive. Perhaps using `allow_fallback: bool = ...` would be better. I think monospacing the return type would be good too.

The one thing I notice on the Pylance screenshot is that the documentation seems to be on another section of the tooltip. What you have is fine.

**Comment by JCWasmx86** [comment]
> Does Meson not document what the defaults are for optional kwargs? 

Yes. It requires again copy-pasting from the docs. For the long run, it would be cool, if even the modules would have everything in that yaml file meson can generate as it could drop a lot of code here. (Or make it autogenerated) I've applied your other suggestions

**Comment by tristan957** [comment]
> Yes

We definitely need to change this.
